app/models/learning_hour.rb

```diff
@@ -10,6 +10,21 @@ class LearningHour < ApplicationRecord
   validate :occurred_at_not_in_future
   validates :learning_hour_topic, presence: true, if: :user_org_learning_topic_enable?
 
+  scope :supervisor_volunteers_learning_hours, ->(supervisor_id) {
+    joins(user: :supervisor_volunteer)
+      .where(supervisor_volunteers: {supervisor_id: supervisor_id})
+      .select("users.id as user_id, users.display_name, SUM(learning_hours.duration_minutes + learning_hours.duration_hours * 60) AS total_time_spent")
+      .group("users.display_name, users.id")
+      .order("users.display_name")
+  }
+
+  scope :all_volunteers_learning_hours, -> {
+    joins(:user)
+      .select("users.id as user_id, users.display_name, SUM(learning_hours.duration_minutes + learning_hours.duration_hours * 60) AS total_time_spent")
+      .group("users.display_name, users.id")
+      .order("users.display_name")
+  }
```

Comment:
I’m worried that these queries are going to display information about users who belong to a different organization. Can you update your model specs and include a user with learning hours who belongs to a different organization and make sure those hours do not get returned?

Reply:
My bad I missed to check organization for the users. Have updated code and spec to include only organization as of the Admin to give correct result